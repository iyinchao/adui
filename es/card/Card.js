function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { forwardRef } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Header from "./Header";
import "./style";
var prefix = "adui-card";
var Card = forwardRef(function (_ref, ref) {
  var children = _ref.children,
      className = _ref.className,
      elevation = _ref.elevation,
      interactive = _ref.interactive,
      otherProps = _objectWithoutProperties(_ref, ["children", "className", "elevation", "interactive"]);

  var classSet = classNames(className, "".concat(prefix, "-base"), "".concat(prefix, "-elevation_").concat(elevation), _defineProperty({}, "".concat(prefix, "-interactive"), interactive));
  return React.createElement("div", _extends({
    ref: ref,
    className: classSet
  }, otherProps), children);
});
Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  elevation: PropTypes.oneOf([0, 1, 2, 3, 4]),
  interactive: PropTypes.bool
};
Card.defaultProps = {
  children: "",
  className: undefined,
  elevation: 1,
  interactive: false
};
Card.Header = Header;
export default Card;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL2NvbXBvbmVudHMvY2FyZC9DYXJkLnRzeCJdLCJuYW1lcyI6WyJSZWFjdCIsImZvcndhcmRSZWYiLCJQcm9wVHlwZXMiLCJjbGFzc05hbWVzIiwiSGVhZGVyIiwicHJlZml4IiwiQ2FyZCIsInJlZiIsImNoaWxkcmVuIiwiY2xhc3NOYW1lIiwiZWxldmF0aW9uIiwiaW50ZXJhY3RpdmUiLCJvdGhlclByb3BzIiwiY2xhc3NTZXQiLCJwcm9wVHlwZXMiLCJub2RlIiwic3RyaW5nIiwib25lT2YiLCJib29sIiwiZGVmYXVsdFByb3BzIiwidW5kZWZpbmVkIl0sIm1hcHBpbmdzIjoiOzs7Ozs7OztBQUFBLE9BQU9BLEtBQVAsSUFBZ0JDLFVBQWhCLFFBQWtDLE9BQWxDO0FBQ0EsT0FBT0MsU0FBUCxNQUFzQixZQUF0QjtBQUNBLE9BQU9DLFVBQVAsTUFBdUIsWUFBdkI7QUFDQSxPQUFPQyxNQUFQLE1BQW1CLFVBQW5CO0FBQ0EsT0FBTyxTQUFQO0FBRUEsSUFBTUMsTUFBTSxHQUFHLFdBQWY7QUFpQ0EsSUFBTUMsSUFBVyxHQUFHTCxVQUFVLENBQzVCLGdCQUVFTSxHQUZGLEVBR0s7QUFBQSxNQUZEQyxRQUVDLFFBRkRBLFFBRUM7QUFBQSxNQUZTQyxTQUVULFFBRlNBLFNBRVQ7QUFBQSxNQUZvQkMsU0FFcEIsUUFGb0JBLFNBRXBCO0FBQUEsTUFGK0JDLFdBRS9CLFFBRitCQSxXQUUvQjtBQUFBLE1BRitDQyxVQUUvQzs7QUFDSCxNQUFNQyxRQUFRLEdBQUdWLFVBQVUsQ0FDekJNLFNBRHlCLFlBRXRCSixNQUZzQixzQkFHdEJBLE1BSHNCLHdCQUdGSyxTQUhFLGlDQUtuQkwsTUFMbUIsbUJBS0lNLFdBTEosRUFBM0I7QUFRQSxTQUNFO0FBQUssSUFBQSxHQUFHLEVBQUVKLEdBQVY7QUFBZSxJQUFBLFNBQVMsRUFBRU07QUFBMUIsS0FBd0NELFVBQXhDLEdBQ0dKLFFBREgsQ0FERjtBQUtELENBbEIyQixDQUE5QjtBQW9CQUYsSUFBSSxDQUFDUSxTQUFMLEdBQWlCO0FBSWZOLEVBQUFBLFFBQVEsRUFBRU4sU0FBUyxDQUFDYSxJQUpMO0FBUWZOLEVBQUFBLFNBQVMsRUFBRVAsU0FBUyxDQUFDYyxNQVJOO0FBWWZOLEVBQUFBLFNBQVMsRUFBRVIsU0FBUyxDQUFDZSxLQUFWLENBQWdCLENBQUMsQ0FBRCxFQUFJLENBQUosRUFBTyxDQUFQLEVBQVUsQ0FBVixFQUFhLENBQWIsQ0FBaEIsQ0FaSTtBQWdCZk4sRUFBQUEsV0FBVyxFQUFFVCxTQUFTLENBQUNnQjtBQWhCUixDQUFqQjtBQW1CQVosSUFBSSxDQUFDYSxZQUFMLEdBQW9CO0FBQ2xCWCxFQUFBQSxRQUFRLEVBQUUsRUFEUTtBQUVsQkMsRUFBQUEsU0FBUyxFQUFFVyxTQUZPO0FBR2xCVixFQUFBQSxTQUFTLEVBQUUsQ0FITztBQUlsQkMsRUFBQUEsV0FBVyxFQUFFO0FBSkssQ0FBcEI7QUFPQUwsSUFBSSxDQUFDRixNQUFMLEdBQWNBLE1BQWQ7QUFFQSxlQUFlRSxJQUFmIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IFJlYWN0LCB7IGZvcndhcmRSZWYgfSBmcm9tIFwicmVhY3RcIlxuaW1wb3J0IFByb3BUeXBlcyBmcm9tIFwicHJvcC10eXBlc1wiXG5pbXBvcnQgY2xhc3NOYW1lcyBmcm9tIFwiY2xhc3NuYW1lc1wiXG5pbXBvcnQgSGVhZGVyIGZyb20gXCIuL0hlYWRlclwiXG5pbXBvcnQgXCIuL3N0eWxlXCJcblxuY29uc3QgcHJlZml4ID0gXCJhZHVpLWNhcmRcIlxuXG5leHBvcnQgaW50ZXJmYWNlIElDYXJkUHJvcHMge1xuICBba2V5OiBzdHJpbmddOiBhbnlcbiAgLyoqXG4gICAqIOWtkOiKgueCuVxuICAgKi9cbiAgY2hpbGRyZW4/OiBSZWFjdC5SZWFjdE5vZGVcbiAgLyoqXG4gICAqIOmZhOWKoOexu+WQjVxuICAgKi9cbiAgY2xhc3NOYW1lPzogc3RyaW5nXG4gIC8qKlxuICAgKiDlsYLnuqdcbiAgICovXG4gIGVsZXZhdGlvbj86IDAgfCAxIHwgMiB8IDMgfCA0XG4gIC8qKlxuICAgKiDlj69cYuS6kuWKqOaAp1xuICAgKi9cbiAgaW50ZXJhY3RpdmU/OiBib29sZWFuXG59XG5cbmV4cG9ydCBpbnRlcmZhY2UgSUNhcmRcbiAgZXh0ZW5kcyBSZWFjdC5Gb3J3YXJkUmVmRXhvdGljQ29tcG9uZW50PFxuICAgIElDYXJkUHJvcHMgJiBSZWFjdC5SZWZBdHRyaWJ1dGVzPEhUTUxEaXZFbGVtZW50PlxuICA+IHtcbiAgSGVhZGVyOiB0eXBlb2YgSGVhZGVyXG59XG5cbi8qKlxuICog5Y2h54mH5piv5YyF5ZCr5LiA57G754m55a6a5pWw5o2u6ZuG55qE5pyJ5bGC57qn55qE5qih5Z2X77yM55So5Lul5om/6L2955u45YWz6IGU5YaF5a6544CC6aG16Z2i6K6+6K6h5Lit77yM6YCa5bi45Lul5YaF5a655Yaz5a6a5Y2h54mH55qE5biD5bGA5pa55byP44CCXG4gKi9cbi8vIEB0cy1pZ25vcmVcbmNvbnN0IENhcmQ6IElDYXJkID0gZm9yd2FyZFJlZihcbiAgKFxuICAgIHsgY2hpbGRyZW4sIGNsYXNzTmFtZSwgZWxldmF0aW9uLCBpbnRlcmFjdGl2ZSwgLi4ub3RoZXJQcm9wcyB9OiBJQ2FyZFByb3BzLFxuICAgIHJlZlxuICApID0+IHtcbiAgICBjb25zdCBjbGFzc1NldCA9IGNsYXNzTmFtZXMoXG4gICAgICBjbGFzc05hbWUsXG4gICAgICBgJHtwcmVmaXh9LWJhc2VgLFxuICAgICAgYCR7cHJlZml4fS1lbGV2YXRpb25fJHtlbGV2YXRpb259YCxcbiAgICAgIHtcbiAgICAgICAgW2Ake3ByZWZpeH0taW50ZXJhY3RpdmVgXTogaW50ZXJhY3RpdmUsXG4gICAgICB9XG4gICAgKVxuICAgIHJldHVybiAoXG4gICAgICA8ZGl2IHJlZj17cmVmfSBjbGFzc05hbWU9e2NsYXNzU2V0fSB7Li4ub3RoZXJQcm9wc30+XG4gICAgICAgIHtjaGlsZHJlbn1cbiAgICAgIDwvZGl2PlxuICAgIClcbiAgfVxuKVxuQ2FyZC5wcm9wVHlwZXMgPSB7XG4gIC8qKlxuICAgKiDlrZDoioLngrlcbiAgICovXG4gIGNoaWxkcmVuOiBQcm9wVHlwZXMubm9kZSxcbiAgLyoqXG4gICAqIOmZhOWKoOexu+WQjVxuICAgKi9cbiAgY2xhc3NOYW1lOiBQcm9wVHlwZXMuc3RyaW5nLFxuICAvKipcbiAgICog5bGC57qnXG4gICAqL1xuICBlbGV2YXRpb246IFByb3BUeXBlcy5vbmVPZihbMCwgMSwgMiwgMywgNF0pLFxuICAvKipcbiAgICog5Y+vXGLkupLliqjmgKdcbiAgICovXG4gIGludGVyYWN0aXZlOiBQcm9wVHlwZXMuYm9vbCxcbn1cblxuQ2FyZC5kZWZhdWx0UHJvcHMgPSB7XG4gIGNoaWxkcmVuOiBcIlwiLFxuICBjbGFzc05hbWU6IHVuZGVmaW5lZCxcbiAgZWxldmF0aW9uOiAxLFxuICBpbnRlcmFjdGl2ZTogZmFsc2UsXG59XG5cbkNhcmQuSGVhZGVyID0gSGVhZGVyXG5cbmV4cG9ydCBkZWZhdWx0IENhcmRcbiJdfQ==