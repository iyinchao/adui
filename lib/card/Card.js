"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Header = _interopRequireDefault(require("./Header"));

require("./style");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var prefix = "adui-card";
var Card = (0, _react.forwardRef)(function (_ref, ref) {
  var children = _ref.children,
      className = _ref.className,
      elevation = _ref.elevation,
      interactive = _ref.interactive,
      otherProps = _objectWithoutProperties(_ref, ["children", "className", "elevation", "interactive"]);

  var classSet = (0, _classnames["default"])(className, "".concat(prefix, "-base"), "".concat(prefix, "-elevation_").concat(elevation), _defineProperty({}, "".concat(prefix, "-interactive"), interactive));
  return _react["default"].createElement("div", _extends({
    ref: ref,
    className: classSet
  }, otherProps), children);
});
Card.propTypes = {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  elevation: _propTypes["default"].oneOf([0, 1, 2, 3, 4]),
  interactive: _propTypes["default"].bool
};
Card.defaultProps = {
  children: "",
  className: undefined,
  elevation: 1,
  interactive: false
};
Card.Header = _Header["default"];
var _default = Card;
exports["default"] = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL2NvbXBvbmVudHMvY2FyZC9DYXJkLnRzeCJdLCJuYW1lcyI6WyJwcmVmaXgiLCJDYXJkIiwicmVmIiwiY2hpbGRyZW4iLCJjbGFzc05hbWUiLCJlbGV2YXRpb24iLCJpbnRlcmFjdGl2ZSIsIm90aGVyUHJvcHMiLCJjbGFzc1NldCIsInByb3BUeXBlcyIsIlByb3BUeXBlcyIsIm5vZGUiLCJzdHJpbmciLCJvbmVPZiIsImJvb2wiLCJkZWZhdWx0UHJvcHMiLCJ1bmRlZmluZWQiLCJIZWFkZXIiXSwibWFwcGluZ3MiOiI7Ozs7Ozs7OztBQUFBOztBQUNBOztBQUNBOztBQUNBOztBQUNBOzs7Ozs7Ozs7Ozs7Ozs7O0FBRUEsSUFBTUEsTUFBTSxHQUFHLFdBQWY7QUFpQ0EsSUFBTUMsSUFBVyxHQUFHLHVCQUNsQixnQkFFRUMsR0FGRixFQUdLO0FBQUEsTUFGREMsUUFFQyxRQUZEQSxRQUVDO0FBQUEsTUFGU0MsU0FFVCxRQUZTQSxTQUVUO0FBQUEsTUFGb0JDLFNBRXBCLFFBRm9CQSxTQUVwQjtBQUFBLE1BRitCQyxXQUUvQixRQUYrQkEsV0FFL0I7QUFBQSxNQUYrQ0MsVUFFL0M7O0FBQ0gsTUFBTUMsUUFBUSxHQUFHLDRCQUNmSixTQURlLFlBRVpKLE1BRlksc0JBR1pBLE1BSFksd0JBR1FLLFNBSFIsaUNBS1RMLE1BTFMsbUJBS2NNLFdBTGQsRUFBakI7QUFRQSxTQUNFO0FBQUssSUFBQSxHQUFHLEVBQUVKLEdBQVY7QUFBZSxJQUFBLFNBQVMsRUFBRU07QUFBMUIsS0FBd0NELFVBQXhDLEdBQ0dKLFFBREgsQ0FERjtBQUtELENBbEJpQixDQUFwQjtBQW9CQUYsSUFBSSxDQUFDUSxTQUFMLEdBQWlCO0FBSWZOLEVBQUFBLFFBQVEsRUFBRU8sc0JBQVVDLElBSkw7QUFRZlAsRUFBQUEsU0FBUyxFQUFFTSxzQkFBVUUsTUFSTjtBQVlmUCxFQUFBQSxTQUFTLEVBQUVLLHNCQUFVRyxLQUFWLENBQWdCLENBQUMsQ0FBRCxFQUFJLENBQUosRUFBTyxDQUFQLEVBQVUsQ0FBVixFQUFhLENBQWIsQ0FBaEIsQ0FaSTtBQWdCZlAsRUFBQUEsV0FBVyxFQUFFSSxzQkFBVUk7QUFoQlIsQ0FBakI7QUFtQkFiLElBQUksQ0FBQ2MsWUFBTCxHQUFvQjtBQUNsQlosRUFBQUEsUUFBUSxFQUFFLEVBRFE7QUFFbEJDLEVBQUFBLFNBQVMsRUFBRVksU0FGTztBQUdsQlgsRUFBQUEsU0FBUyxFQUFFLENBSE87QUFJbEJDLEVBQUFBLFdBQVcsRUFBRTtBQUpLLENBQXBCO0FBT0FMLElBQUksQ0FBQ2dCLE1BQUwsR0FBY0Esa0JBQWQ7ZUFFZWhCLEkiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgUmVhY3QsIHsgZm9yd2FyZFJlZiB9IGZyb20gXCJyZWFjdFwiXG5pbXBvcnQgUHJvcFR5cGVzIGZyb20gXCJwcm9wLXR5cGVzXCJcbmltcG9ydCBjbGFzc05hbWVzIGZyb20gXCJjbGFzc25hbWVzXCJcbmltcG9ydCBIZWFkZXIgZnJvbSBcIi4vSGVhZGVyXCJcbmltcG9ydCBcIi4vc3R5bGVcIlxuXG5jb25zdCBwcmVmaXggPSBcImFkdWktY2FyZFwiXG5cbmV4cG9ydCBpbnRlcmZhY2UgSUNhcmRQcm9wcyB7XG4gIFtrZXk6IHN0cmluZ106IGFueVxuICAvKipcbiAgICog5a2Q6IqC54K5XG4gICAqL1xuICBjaGlsZHJlbj86IFJlYWN0LlJlYWN0Tm9kZVxuICAvKipcbiAgICog6ZmE5Yqg57G75ZCNXG4gICAqL1xuICBjbGFzc05hbWU/OiBzdHJpbmdcbiAgLyoqXG4gICAqIOWxgue6p1xuICAgKi9cbiAgZWxldmF0aW9uPzogMCB8IDEgfCAyIHwgMyB8IDRcbiAgLyoqXG4gICAqIOWPr1xi5LqS5Yqo5oCnXG4gICAqL1xuICBpbnRlcmFjdGl2ZT86IGJvb2xlYW5cbn1cblxuZXhwb3J0IGludGVyZmFjZSBJQ2FyZFxuICBleHRlbmRzIFJlYWN0LkZvcndhcmRSZWZFeG90aWNDb21wb25lbnQ8XG4gICAgSUNhcmRQcm9wcyAmIFJlYWN0LlJlZkF0dHJpYnV0ZXM8SFRNTERpdkVsZW1lbnQ+XG4gID4ge1xuICBIZWFkZXI6IHR5cGVvZiBIZWFkZXJcbn1cblxuLyoqXG4gKiDljaHniYfmmK/ljIXlkKvkuIDnsbvnibnlrprmlbDmja7pm4bnmoTmnInlsYLnuqfnmoTmqKHlnZfvvIznlKjku6Xmib/ovb3nm7jlhbPogZTlhoXlrrnjgILpobXpnaLorr7orqHkuK3vvIzpgJrluLjku6XlhoXlrrnlhrPlrprljaHniYfnmoTluIPlsYDmlrnlvI/jgIJcbiAqL1xuLy8gQHRzLWlnbm9yZVxuY29uc3QgQ2FyZDogSUNhcmQgPSBmb3J3YXJkUmVmKFxuICAoXG4gICAgeyBjaGlsZHJlbiwgY2xhc3NOYW1lLCBlbGV2YXRpb24sIGludGVyYWN0aXZlLCAuLi5vdGhlclByb3BzIH06IElDYXJkUHJvcHMsXG4gICAgcmVmXG4gICkgPT4ge1xuICAgIGNvbnN0IGNsYXNzU2V0ID0gY2xhc3NOYW1lcyhcbiAgICAgIGNsYXNzTmFtZSxcbiAgICAgIGAke3ByZWZpeH0tYmFzZWAsXG4gICAgICBgJHtwcmVmaXh9LWVsZXZhdGlvbl8ke2VsZXZhdGlvbn1gLFxuICAgICAge1xuICAgICAgICBbYCR7cHJlZml4fS1pbnRlcmFjdGl2ZWBdOiBpbnRlcmFjdGl2ZSxcbiAgICAgIH1cbiAgICApXG4gICAgcmV0dXJuIChcbiAgICAgIDxkaXYgcmVmPXtyZWZ9IGNsYXNzTmFtZT17Y2xhc3NTZXR9IHsuLi5vdGhlclByb3BzfT5cbiAgICAgICAge2NoaWxkcmVufVxuICAgICAgPC9kaXY+XG4gICAgKVxuICB9XG4pXG5DYXJkLnByb3BUeXBlcyA9IHtcbiAgLyoqXG4gICAqIOWtkOiKgueCuVxuICAgKi9cbiAgY2hpbGRyZW46IFByb3BUeXBlcy5ub2RlLFxuICAvKipcbiAgICog6ZmE5Yqg57G75ZCNXG4gICAqL1xuICBjbGFzc05hbWU6IFByb3BUeXBlcy5zdHJpbmcsXG4gIC8qKlxuICAgKiDlsYLnuqdcbiAgICovXG4gIGVsZXZhdGlvbjogUHJvcFR5cGVzLm9uZU9mKFswLCAxLCAyLCAzLCA0XSksXG4gIC8qKlxuICAgKiDlj69cYuS6kuWKqOaAp1xuICAgKi9cbiAgaW50ZXJhY3RpdmU6IFByb3BUeXBlcy5ib29sLFxufVxuXG5DYXJkLmRlZmF1bHRQcm9wcyA9IHtcbiAgY2hpbGRyZW46IFwiXCIsXG4gIGNsYXNzTmFtZTogdW5kZWZpbmVkLFxuICBlbGV2YXRpb246IDEsXG4gIGludGVyYWN0aXZlOiBmYWxzZSxcbn1cblxuQ2FyZC5IZWFkZXIgPSBIZWFkZXJcblxuZXhwb3J0IGRlZmF1bHQgQ2FyZFxuIl19