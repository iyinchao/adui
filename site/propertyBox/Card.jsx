import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Slider, Switch, Button, Card } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    size: "small",
    header: true,
    title: true,
    subTitle: false,
    interactive: false,
    elevation: 1,
  }

  render() {
    const { size, header, title, subTitle, interactive, elevation } = this.state

    const codes = `<Card
  ${interactive ? `interactive` : ""}
  elevation={${elevation}}
>
  ${
    header
      ? `<Card.Header 
    ${title ? `title="卡片标题"` : ""} 
    ${size === "small" ? "" : `size="${size}"`}
    ${subTitle ? `subTitle="副标题"` : ""} 
    ${title ? `style={{ boxShadow: "0 1px 0 0 rgba(0,0,0,0.06)" }}` : ""}
  />`
      : ""
  }
  <div> ... </div>
</Card>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div
            className={styles.left}
            style={{ padding: "40px 90px", backgroundColor: "#fafafa" }}
          >
            <Card
              style={{ flex: 1 }}
              interactive={interactive}
              elevation={elevation}
            >
              {header && (
                <Card.Header
                  title={title ? "卡片标题" : ""}
                  size={size}
                  subTitle={subTitle ? "副标题" : ""}
                  style={{ boxShadow: "0 1px 0 0 rgba(0, 0, 0, 0.06)" }}
                />
              )}
              <div style={{ height: "182px" }} />
            </Card>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
              <dt className={styles.controlTitle}>投影层级</dt>
              <dd className={styles.controlContent}>
                <Slider
                  inputVisible={false}
                  defaultValue={1}
                  onChange={(value) => {
                    this.setState({ elevation: value })
                  }}
                  min={0}
                  max={4}
                  marks={{ 0: 0, 1: 1, 2: 2, 3: 3, 4: 4 }}
                />
              </dd>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="卡片顶栏"
                  unCheckedText="卡片顶栏"
                  checked={header}
                  onChange={(param) => this.setState({ header: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="标题"
                  unCheckedText="标题"
                  checked={title}
                  onChange={(param) => this.setState({ title: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="副标题"
                  unCheckedText="副标题"
                  checked={subTitle}
                  onChange={(param) => this.setState({ subTitle: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="可交互"
                  unCheckedText="可交互"
                  checked={interactive}
                  onChange={(param) => this.setState({ interactive: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
