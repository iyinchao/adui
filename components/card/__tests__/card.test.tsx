import * as React from "react"
import TestRenderer from "react-test-renderer"
import Card from ".."

describe("Card", () => {
  it("Card 是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Card>
        <Card.Header title="title" subTitle="title">
          children
        </Card.Header>
        children
      </Card>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
