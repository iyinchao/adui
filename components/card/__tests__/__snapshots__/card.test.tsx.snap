// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card 是否正确渲染 1`] = `
<div
  className="adui-card-base adui-card-elevation_1"
>
  <div
    className="adui-card-header adui-card-small"
  >
    <div
      className="adui-card-headerBlock"
    >
      <div
        className="adui-card-headerTop"
      >
        <div
          className="adui-card-title"
        >
          title
        </div>
        <div
          className="adui-card-subTitle"
        >
          title
        </div>
      </div>
    </div>
    <div
      className="adui-card-children"
    >
      children
    </div>
  </div>
  children
</div>
`;
